modules/example/src/smithy4s/example/StructureWithRefinedTypes.scala

```diff
@@ -4,17 +4,20 @@ import smithy4s.Hints
 import smithy4s.Schema
 import smithy4s.ShapeId
 import smithy4s.ShapeTag
+import smithy4s.example.refined.Age
+import smithy4s.example.refined.Age.provider._
+import smithy4s.schema.Schema.int
 import smithy4s.schema.Schema.struct
 
-case class StructureWithRefinedTypes(age: Age, personAge: PersonAge, fancyList: Option[smithy4s.example.FancyList] = None, unwrappedFancyList: Option[smithy4s.example.refined.FancyList] = None, name: Option[smithy4s.example.Name] = None, dogName: Option[smithy4s.example.refined.Name] = None)
+case class StructureWithRefinedTypes(age: Option[Age] = None, personAge: Option[Age] = None, fancyList: Option[smithy4s.example.FancyList] = None, unwrappedFancyList: Option[smithy4s.example.refined.FancyList] = None, name: Option[smithy4s.example.Name] = None, dogName: Option[smithy4s.example.refined.Name] = None)
```

Comment:
interesting: there was no option before, even though it was not `required`

Reply:
That's most likely because the `$version` clause is not defined in the smithy file, and the smithy 2.0 parser/assembler automatically adds `default` traits in some conditions when parsing smithy 1.0 files. One condition might be the absence of other traits.

Reply:
I decided to try a few different things to test, but my observations are as follow:

![Screenshot 2023-02-02 at 10 15 55](https://user-images.githubusercontent.com/5230460/216364318-445e119e-e66d-42fb-a5f2-3e7904e97d30.png)

All cases have either a default or are using `Option` except the one where `ageFormat` is applied to a custom simple shape like `integer Age`.

Reply:
tbh, I'm having difficulties digesting this table without seeing the smithy code. Also, the information that's missing is what the model tells us once it's been loaded : what is the condition for having an automatically applied `@default` trait ?

Reply:
So your assumptions when you said:
> That's most likely because the $version clause is not defined in the smithy file, and the smithy 2.0 parser/assembler automatically adds default traits

Is absolutely true, but it does not seem to be related to traits but instead is just because the type is `integer`: https://smithy.io/2.0/guides/migrating-idl-1-to-2.html?highlight=boxed#converting-primitive-root-level-shapes-from-1-0 or in [code](https://github.com/awslabs/smithy/blob/56263d64885688118225acfba565d5eea6d1bc3c/smithy-model/src/main/java/software/amazon/smithy/model/loader/ModelInteropTransformer.java#L111)

And the [full list of types that have default is here](https://github.com/awslabs/smithy/blob/56263d64885688118225acfba565d5eea6d1bc3c/smithy-model/src/main/java/software/amazon/smithy/model/loader/ModelInteropTransformer.java#L62).

But I feel like the `default` value supplied does not play well with the refinement type. For all we know, the refinement provider could consider `0` to be an invalid age.

Reply:
Agreed. I wonder what happens if you put something like `@range(min: 1)` on top of the integer... does the default get generated still ?

Reply:
yes, haha: https://github.com/daddykotex/scala-scripts/blob/main/smithy-playground.scala#L38